LivlReviews.Api/Controllers/ProductController.cs

```diff
@@ -0,0 +1,21 @@
+using LivlReviews.Domain.Entities;
+using LivlReviews.Domain.Models;
+using LivlReviews.Infra.Repositories.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LivlReviews.Api.Controllers;
+
+[Authorize]
+[ApiController]
+[Route("[controller]")]
+public class ProductController(IPaginatedRepository<Product> repository) : ControllerBase
+{
+    [HttpGet]
+    public ActionResult<PaginatedResult<Product>> GetProducts(int page = 1, int pageSize = 10, string search = "")
+    {
+        PaginationParameters paginationParameters = new PaginationParameters { page = page, pageSize = pageSize };
+        PaginatedResult<Product> products = repository.GetPaginated(product => product.Name.Contains(search), paginationParameters);
```

Comment:
As said by mister topcu here at the end, if I have well understood: 
https://youtu.be/-dXN8wkN0yk?si=5WgWFT0vfVpuFRU4&t=2677

We should not directly use Product in our Api project, but instead something like "ProductRessource" that is mapping the Product properties in case we rename things, to do not break it on the client side...

While I do not thing this is necessary for the size of our project, should be implement such a thing ? For the beautifull eyes of Senior Edouardo ? (and ours eyes too)

Feel free to close this comment if we don't give a fuckk :D